src/commands.c

```diff
@@ -6362,14 +6372,25 @@ void lastscore_add()
 			}
 		}
 	}
-	else if ((isTeam() || isCTF()) && k_showscores)
+	else if ((isTeam() || isCTF() || isCA()) && k_showscores)
 	{
-		lst = isTeam() ? lsTeam : lsCTF;
+		if (isCA())
+		{
+			lst = cvar("k_clan_arena") == 2 ? lsWO : lsCA;
+		}
+		else if (isTeam())
+		{
+			lst = lsTeam;
+		}
+		else
+		{
+			lst = lsCTF;
+		}
 
 		e1 = cvar_string("_k_team1");
-		s1 = get_scores1();
+		s1 = isCA() ? CA_get_score_1() : get_scores1();
 		e2 = cvar_string("_k_team2");
-		s2 = get_scores2();
+		s2 = isCA() ? CA_get_score_2() : get_scores2();
 
```

Comment:
Here we also have 3 consecutive calls of `isCA()`. Should we not optimize this a bit:
```
	else if ((isTeam() || isCTF() || isCA()) && k_showscores)
	{
		qbool isCa = isCA();
		if (isCa)
		{
			lst = cvar("k_clan_arena") == 2 ? lsWO : lsCA;
		}
		else if (isTeam())
		{
			lst = lsTeam;
		}
		else
		{
			lst = lsCTF;
		}

		e1 = cvar_string("_k_team1");
		s1 = isCa ? CA_get_score_1() : get_scores1();
		e2 = cvar_string("_k_team2");
		s2 = isCa ? CA_get_score_2() : get_scores2();

		// players from first team
		for (t1[0] = from = 0, p = world; (p = find_plrghst(p, &from));)
```


or at least:
```
		e1 = cvar_string("_k_team1");
		e2 = cvar_string("_k_team2");
		if (isCA())
		{
				s1 = CA_get_score_1();
				s2 = CA_get_score_2();
		}
		else
		{
				s1 = get_scores1();
				s2 = get_scores2();
		}
```

Reply:
changed.